# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1367113548 14400
#      Sat Apr 27 21:45:48 2013 -0400
# Node ID 1a1f8199b5bb73321718ceccbff5c07ec0f17ddc
# Parent  b678d2d10e53441e9755224fadcd65238c54559d
generate single-line commands from workspace view context menu

* workspace-view.cc (workspace_view::handle_contextmenu_plot,
workspace_view::handle_contextmenu_stem): Don't insert newline after
figure command.
(workspace_view::handle_contextmenu_command): Don't append newline to
command string.

diff --git a/libgui/src/workspace-view.cc b/libgui/src/workspace-view.cc
--- a/libgui/src/workspace-view.cc
+++ b/libgui/src/workspace-view.cc
@@ -191,29 +191,29 @@ workspace_view::handle_contextmenu_renam
       if (ok && ! new_name.isEmpty ())
         m->setData (index, new_name, Qt::EditRole);
     }
 }
 
 void
 workspace_view::handle_contextmenu_disp (void)
 {
-  relay_contextmenu_command ("disp"); 
+  relay_contextmenu_command ("disp");
 }
 
 void
 workspace_view::handle_contextmenu_plot (void)
 {
-  relay_contextmenu_command ("figure;\nplot"); 
+  relay_contextmenu_command ("figure (); plot");
 }
 
 void
 workspace_view::handle_contextmenu_stem (void)
 {
-  relay_contextmenu_command ("figure;\nstem"); 
+  relay_contextmenu_command ("figure (); stem");
 }
 
 void
 workspace_view::relay_contextmenu_command (const QString& cmdname)
 {
   QModelIndex index = view->currentIndex ();
 
   if (index.isValid ())
@@ -221,17 +221,17 @@ workspace_view::relay_contextmenu_comman
       index = index.sibling(index.row(), 0);
 
       QAbstractItemModel *m = view->model ();
 
       QMap<int, QVariant> item_data = m->itemData (index);
   
       QString var_name = item_data[0].toString ();
 
-      emit command_requested (cmdname + "(" + var_name + ")\n");
+      emit command_requested (cmdname + " (" + var_name + ");");
     }
 }
 
 void
 workspace_view::handle_model_changed (void)
 {
   // Just modify those rows that have been added rather than go through
   // the whole list.  For-loop test will handle when number of rows reduced.
